projects/driver_display/src/driver_display_brightness.c

```diff
@@ -0,0 +1,80 @@
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "crc32.h"
+#include "gpio.h"
+#include "log.h"
+#include "misc.h"
+#include "persist.h"
+#include "soft_timer.h"
+#include "status.h"
+
+#include "driver_display_brightness.h"
+#include "driver_display_config.h"
+
+static PersistStorage s_persist;
+static DriverDisplayBrightnessData data;
+
+void driver_display_brightness_timer_callback(SoftTimerID timer_id, void *context) {
+  flash_erase(DRIVER_DISPLAY_CONFIG_PERSIST_PAGE);
+  persist_init(&s_persist, DRIVER_DISPLAY_CONFIG_PERSIST_PAGE, &data, sizeof(data));
```

Comment:
You should not be initializing the persist layer in a callback - in fact, you shouldn't be using a persist layer at all.